rnd/autogpt_builder/src/app/page.tsx

```diff
@@ -17,62 +18,53 @@ import {
 } from "@/components/monitor";
 
 const Monitor = () => {
-  const [flows, setFlows] = useState<GraphMeta[]>([]);
-  const [flowRuns, setFlowRuns] = useState<FlowRun[]>([]);
+  const [agents, setAgents] = useState<GraphMeta[]>([]);
+  const [agentRuns, setAgentRuns] = useState<FlowRun[]>([]);
   const [selectedFlow, setSelectedFlow] = useState<GraphMeta | null>(null);
   const [selectedRun, setSelectedRun] = useState<FlowRun | null>(null);
 
   const api = useMemo(() => new AutoGPTServerAPI(), []);
 
-  const refreshFlowRuns = useCallback(
-    (flowID: string) => {
-      // Fetch flow run IDs
-      api.listGraphRunIDs(flowID).then((runIDs) =>
-        runIDs.map((runID) => {
-          let run;
-          if (
-            (run = flowRuns.find((fr) => fr.id == runID)) &&
-            !["waiting", "running"].includes(run.status)
-          ) {
-            return;
-          }
-
-          // Fetch flow run
-          api.getGraphExecutionInfo(flowID, runID).then((execInfo) =>
-            setFlowRuns((flowRuns) => {
-              if (execInfo.length == 0) return flowRuns;
-
-              const flowRunIndex = flowRuns.findIndex((fr) => fr.id == runID);
-              const flowRun = flowRunFromNodeExecutionResults(execInfo);
-              if (flowRunIndex > -1) {
-                flowRuns.splice(flowRunIndex, 1, flowRun);
-              } else {
-                flowRuns.push(flowRun);
-              }
-              return [...flowRuns];
-            }),
-          );
-        }),
+  const fetchAgents = useCallback(() => {
+    api.listGraphs().then((agent) => {
+      setAgents(agent);
+      const flowRuns = agent.flatMap((graph) =>
+        graph.executions != null
+          ? graph.executions.map((execution) => ({
+              id: execution.execution_id,
+              graphID: graph.id,
+              graphVersion: graph.version,
+              status: execution.status.toLowerCase() as
+                | "running"
+                | "waiting"
+                | "success"
+                | "failed",
+              startTime: execution.started_at.getTime(),
+              endTime: execution.ended_at.getTime(),
+              duration:
+                (execution.ended_at.getTime() -
+                  execution.started_at.getTime()) /
+                1000,
+              totalRunTime:
+                (execution.ended_at.getTime() -
+                  execution.started_at.getTime()) /
+                1000,
+              nodeExecutionResults: [],
+            }))
+          : [],
       );
-    },
-    [api, flowRuns],
-  );
-
-  const fetchFlowsAndRuns = useCallback(() => {
-    api.listGraphs().then((flows) => {
-      setFlows(flows);
-      flows.map((flow) => refreshFlowRuns(flow.id));
+      setAgentRuns(flowRuns);
     });
-  }, [api, refreshFlowRuns]);
+  }, [api]);
+
+  useEffect(() => {
+    fetchAgents();
+  });
```

Comment:
Is this intended? This `useEffect` will run on every render. If you want to run it once:
```suggestion
  useEffect(() => {
    fetchAgents();
  }, []);
```

Reply:
No it was intended to run once on page load

Reply:
I think I've fixed it now?